website/addons/base/views.py

```diff
@@ -300,6 +304,8 @@ def create_waterbutler_log(payload, **kwargs):
 
         node_addon.create_waterbutler_log(auth, action, metadata)
 
+    file_signals.file_updated.send(node=node, user=user, event_type=action, payload=payload)
```

Comment:
If a file event is not in the registry, then this line raises a `RegistryError`. That error is not caught, causes the request to return with an HTTP error, and more importantly - rolls back the transaction. That means that the event is not logged on the OSF side. :(

I recommend that you catch the exception, like so:

```python
try:
    file_signals.file_updated.send(
        node=node,
        user=user,
        event_type=action,
        payload=payload
    )
except RegistryError:
    # make sure we know that an event without a notification type was sent.
```

If I understand the flow correctly, WaterButler is the one making the call to the OSF here, so there's no error presented to the user or the developer if it fails. Perhaps logging the issue to Sentry would be enough, or perhaps it would be justified to email someone.

Another option would be to eliminate the `RegistryError` exception entirely, and instead have a default notification class that notifies us when it's called, instead of bubbling up an exception that must be caught.